tests/jobs/test_scheduler_job.py

```diff
@@ -188,6 +204,41 @@ def test_process_executor_events(self, mock_stats_incr):
 
         mock_stats_incr.assert_called_once_with('scheduler.tasks.killed_externally')
 
+    def test_scheduler_executor_overflow(self):
+        """
+        Test that tasks that tasks that are set back to scheduled and removed from the executor
+        queue in the case of an overflow.
+        """
+        executor = MockExecutor(do_update=True, parallelism=3)
+        with create_session() as session:
+            dag = self.create_test_dag()
+            task = DummyOperator(
+                task_id='dummy',
+                dag=dag,
+                owner='airflow')
+            from datetime import timedelta
+            tis = []
+            for i in range(1, 10):
+                ti = TI(task, DEFAULT_DATE + timedelta(days=i))
+                ti.state = State.SCHEDULED
+                tis.append(ti)
+                session.merge(ti)
+            session.commit()
+
+            simple_dag_bag = self._make_simple_dag_bag([dag])
+            scheduler = SchedulerJob(num_runs=1,
+                                     executor=executor,
+                                     subdir=os.path.join(settings.DAGS_FOLDER,
+                                                         "no_dags.py"))
+            mock.patch.object(scheduler, '_change_state_for_tis_without_dagrun').start()
+            scheduler._process_dags(simple_dag_bag)
+            [ti.refresh_from_db() for ti in tis]
```

Comment:
```suggestion
            for ti in tis:
                ti.refresh_from_db() 
```

as we don't need to create list